packages/api/src/beacon/client/proof.ts

```diff
@@ -1,22 +1,48 @@
 import {IChainForkConfig} from "@lodestar/config";
-import {deserializeProof} from "@chainsafe/persistent-merkle-tree";
+import {deserializeProof, Proof} from "@chainsafe/persistent-merkle-tree";
 import {Api, ReqTypes, routesData, getReqSerializers} from "../routes/proof.js";
-import {IHttpClient, getFetchOptsSerializers} from "../../utils/client/index.js";
+import {IHttpClient, getFetchOptsSerializers, ClientOptions, HttpError} from "../../utils/client/index.js";
+import {HttpStatusCode} from "../../utils/client/httpStatusCode.js";
+import {APIClientResponse} from "../../utils/types.js";
 
 /**
  * REST HTTP client for lightclient routes
  */
-export function getClient(_config: IChainForkConfig, httpClient: IHttpClient): Api {
+export function getClient<ErrorAsResponse extends boolean = false>(
```

Comment:
Can you document the behavior of ErrorAsResponse somewhere?

Seems that depending on the option passed, you can make errors either throw or return.

Do we need both behaviors? Might be simpler to pick a convention (ie always return) and stick with it.